Fix Docker build: shared packages not built in Dockerfile

## Summary

The bloom-web Docker build fails because shared workspace packages (bloom-js, bloom-fs) are not compiled before Next.js build runs.

## Error

```
Type error: Cannot find module '@salk-hpi/bloom-js' or its corresponding type declarations.
```

At Dockerfile.bloom-web.prod line 35: `RUN npm run build`

## Fix

Add shared package build steps before the Next.js build in `web/Dockerfile.bloom-web.prod`:

```dockerfile
RUN cd /app/packages/bloom-js && npx tsc -p tsconfig.json
RUN cd /app/packages/bloom-fs && npx tsc -p tsconfig.json
WORKDIR /app/web
RUN npm run build
```

--- a/web/Dockerfile.bloom-web.prod
+++ b/web/Dockerfile.bloom-web.prod
@@ -30,6 +30,10 @@ RUN npm install
 COPY web ./web
 COPY packages ./packages
 
+# Build shared packages (dist/ is gitignored, must build here)
+RUN cd /app/packages/bloom-js && npx tsc -p tsconfig.json
+RUN cd /app/packages/bloom-fs && npx tsc -p tsconfig.json
+
 # Build the Next.js app
 WORKDIR /app/web
 RUN npm run build

